src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -118,7 +117,8 @@ internal static bool CanGetSemanticInfo(CSharpSyntaxNode node, bool allowNamedAr
                         (node is ExpressionSyntax && (isSpeculative || allowNamedArgumentName || !SyntaxFacts.IsNamedArgumentName(node))) ||
                         (node is ConstructorInitializerSyntax) ||
                         (node is AttributeSyntax) ||
-                        (node is CrefSyntax);
+                        (node is CrefSyntax) ||
+                        (node is DeclarationExpressionSyntax);
```

Comment:
>(node is DeclarationExpressionSyntax) [](start = 24, length = 37)

Do we really need this extra condition? It feels like the very first condition should be good enough.